src/main/steps/applicant1/finalising-your-application/post.ts

```diff
@@ -19,7 +29,37 @@ export default class FinalisingYourApplicationPostController extends PostControl
     return super.save(req, formData, eventName);
   }
 
+  protected getNextUrl(req: AppRequest): string {
+    const hasApplicant2SwitchedToSoleFo =
+      req.session.isApplicant2 &&
+      req.originalUrl === APPLICANT_2 + FINALISING_YOUR_APPLICATION &&
+      req.session.userCase.finalOrderSwitchedToSole === YesOrNo.YES &&
+      this.formData.doesApplicant2WantToApplyForFinalOrder;
+
+    if (hasApplicant2SwitchedToSoleFo) {
+      req.session.isApplicant2 = false;
+      req.originalUrl = FINALISING_YOUR_APPLICATION;
+    }
+
+    return super.getNextUrl(req);
+  }
+
+  protected async saveAndContinue(
+    req: AppRequest<AnyObject>,
+    res: Response,
+    form: Form,
+    formData: Partial<Case>
+  ): Promise<void> {
+    this.formData = formData;
```

Comment:
not sure what this is doing

Reply:
Grabbing the formData and assigning it to the instance of the class so that it's available for the local instance of another method.

Reply:
Refactored and removed the requirement for the formData.